test/Conversion/SCFToCalyx/convert_simple.mlir

```diff
@@ -438,3 +438,72 @@ module {
+// CHECK:      calyx.group @bb0_0 {
+// CHECK-DAG:        calyx.assign %std_compareFN_0.left = %in0 : i64
+// CHECK-DAG:        calyx.assign %std_compareFN_0.right = %in1 : i64
+// CHECK-DAG:        calyx.assign %std_compareFN_0.signaling = %true : i1
+// CHECK-DAG:        calyx.assign %compare_port_0_reg.write_en = %std_compareFN_0.done : i1
+// CHECK-DAG:        calyx.assign %compare_port_0_reg.in = %std_compareFN_0.gt : i1
+// CHECK-DAG:        calyx.assign %unordered_port_0_reg.write_en = %std_compareFN_0.done : i1
+// CHECK-DAG:        calyx.assign %unordered_port_0_reg.in = %std_compareFN_0.unordered : i1
+// CHECK-DAG:        calyx.assign %std_or_0.left = %compare_port_0_reg.out : i1
+// CHECK-DAG:        calyx.assign %std_or_0.right = %unordered_port_0_reg.out : i1
+// CHECK-DAG:        calyx.assign %std_and_0.left = %compare_port_0_reg.done : i1
+// CHECK-DAG:        calyx.assign %std_and_0.right = %unordered_port_0_reg.done : i1
+// CHECK-DAG:        calyx.assign %cmpf_0_reg.in = %std_or_0.out : i1
+// CHECK-DAG:        calyx.assign %cmpf_0_reg.write_en = %std_and_0.out : i1
+// CHECK-DAG:        %0 = comb.xor %std_compareFN_0.done, %true : i1
+// CHECK-DAG:        calyx.assign %std_compareFN_0.go = %0 ? %true : i1
+// CHECK-DAG:        calyx.group_done %cmpf_0_reg.done : i1
+// CHECK-DAG:      }
+// CHECK:      calyx.group @bb0_2 {
+// CHECK-DAG:        calyx.assign %std_compareFN_1.left = %in1 : i64
+// CHECK-DAG:        calyx.assign %std_compareFN_1.right = %in1 : i64
+// CHECK-DAG:        calyx.assign %std_compareFN_1.signaling = %false : i1
+// CHECK-DAG:        calyx.assign %unordered_port_1_reg.write_en = %std_compareFN_1.done : i1
+// CHECK-DAG:        calyx.assign %unordered_port_1_reg.in = %std_compareFN_1.unordered : i1
+// CHECK-DAG:        calyx.assign %cmpf_1_reg.in = %unordered_port_1_reg.out : i1
+// CHECK-DAG:        calyx.assign %cmpf_1_reg.write_en = %unordered_port_1_reg.out : i1
+// CHECK-DAG:        %0 = comb.xor %std_compareFN_1.done, %true : i1
+// CHECK-DAG:        calyx.assign %std_compareFN_1.go = %0 ? %true : i1
+// CHECK-DAG:        calyx.group_done %cmpf_1_reg.done : i1
+// CHECK-DAG:      }
+// CHECK:      calyx.group @ret_assign_0 {
+// CHECK-DAG:        calyx.assign %ret_arg0_reg.in = %std_mux_1.out : i64
+// CHECK-DAG:        calyx.assign %ret_arg0_reg.write_en = %true : i1
+// CHECK-DAG:        calyx.assign %std_mux_1.cond = %cmpf_1_reg.out : i1
+// CHECK-DAG:        calyx.assign %std_mux_1.tru = %in1 : i64
+// CHECK-DAG:        calyx.assign %std_mux_1.fal = %std_mux_0.out : i64
+// CHECK-DAG:        calyx.assign %std_mux_0.cond = %cmpf_0_reg.out : i1
+// CHECK-DAG:        calyx.assign %std_mux_0.tru = %in0 : i64
+// CHECK-DAG:        calyx.assign %std_mux_0.fal = %in1 : i64
+// CHECK-DAG:        calyx.group_done %ret_arg0_reg.done : i1
+// CHECK-DAG:      }
+// CHECK-DAG:    }
+
+module {
+  func.func @main(%arg0: f64, %arg1: f64) -> f64 {
+    %0 = arith.cmpf ugt, %arg0, %arg1 : f64
```

Comment:
Can you add a test for signed integers as well please?

Reply:
I just tried, and strange enough, every operation in the arith dialect either takes floating point or signless integer type. There is even no way to do: `divsi` on two `si64` types..

Reply:
nvm `select` takes `anyType`s